src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -262,6 +268,25 @@ def get(self, request: Request, project: Project) -> Response:
                     else:
                         scope.set_tag("stacktrace_link.error", "file_not_found")
 
+                result["commitSha"] = None
+                githubIntegration = GitHubIntegration(integrations[0], project.organization_id)
```

Comment:
Filter `integrations` to get the one you need rather than doing this. You can then call `get_blame_for_file`.

https://github.com/getsentry/sentry/blob/9b00a291d9dc5cbd63b891283dfd274e32637a4e/src/sentry/api/endpoints/project_stacktrace_link.py#L180